docusaurus/docs/iOS/uikit/components/message.md

```diff
@@ -108,7 +112,29 @@ Components.default.messageLayoutOptionsResolver = CustomMessageLayoutOptionsReso
 | ------------- | ------------- |
 | <img src={require("../../assets/message-basic-resolver-before.png").default} /> | <img src={require("../../assets/message-basic-resolver-after.png").default} /> |
 
-### Date Separators
+#### Decoration Views
+
+The SDK allows you to configure what will be presented above and below your message. Those views we are calling them `ChatMessageDecorationView`. `ChatMessageDecorationView` are fully customizable and they are even used for standard SDK components like the `Date Separators` that we will see below.
+
+In order to provide a `ChatMessageDecorationView` (either a header or a footer) for a message, you will need to implement the following method from the `ChatMessageListVCDelegate`
+```swift
+func chatMessageListVC(
+    _ vc: ChatMessageListVC,
+    decorationViewForMessage message: ChatMessage,
+    decorationType: ChatMessageDecorationType,
+    at indexPath: IndexPath
+) -> ChatMessageDecorationView?
+```
+`ChatMessageDecorationView` are following a similar flow as the `UITableViewHeaderFooterView` decorations for sections that `UITableView` is managing. During the preparation of a message cell, the `ChatMessageListVC` will ask the delegate to provide `ChatMessageDecorationView` for header and footer. If the delegate returns a non-nil value the 
+provided `ChatMessageDecorationView` will be placed in the cell's UI according to it's decoration type. In the case where the delegate will return a nil value then the cell will be updated to release the space for this specific decoration type, if it was previously used.
+
+:::note
+Unlike the `UITableViewHeaderFooterView`, the cell `ChatMessageDecorationView` that the SDK manages, is not cached and reused. This was decided due to some limitations of `UITableViewHeaderFooterView` that were preventing us from making `ChatMessageDecorationView` inherit from.
+
+This is a known limitation and we are working on providing a solution in a following update.
+:::
```

Comment:
We should not provide implementation details in the documentation, especially this one might warn the customer for no reason. 
```suggestion
```